javascript/code-challenges/challenges-13.test.js

```diff
@@ -0,0 +1,318 @@
+CHALLENGE 1 - Review
+
+Write a function named longestString that takes in an array of strings and returns the index position of the longest string.
+------------------------------------------------------------------------------------------------ */
+
+const longestString = (arr) => {
+  let maxStrLength = 0;
+  let maxStrIndex = -1;
+  for (let i = 0; i < arr.length; i++) {
+    if (arr[i].length > maxStrLength) {
+      maxStrLength = arr[i].length;
+      maxStrIndex = i;
+    }
+  }
+  return maxStrIndex;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function named firstLetters that takes in an array of strings and returns an array containing only the first letter of each string.
+
+For example, ['this is great :)', 'wow', 'whyyyyyy :(', ':)))))'] returns ['t', 'w', 'w', ':']
+------------------------------------------------------------------------------------------------ */
+
+const firstLetters = (arr) => {
+  let firstStrArr = [];
+  arr.forEach(str => {
+    firstStrArr.push(str[0]);
+  });
+  return firstStrArr;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 3
+
+Write a function named findHappiness that takes in an array of strings and returns an array containing only the strings from the input array that contain ":)".
+
+For example, ['this is great :)', 'wow', 'whyyyyyy :(', ':)))))'] returns ['this is great :)', ':)))))']
+------------------------------------------------------------------------------------------------ */
+
+const findHappiness = (arr) => {
+  let smileArr = [];
+  arr.forEach(str => {
+    if (str.includes(':)')) {
+      smileArr.push(str);
+    }
+  });
+  return smileArr;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 4
+
+Write a function named standardizePhoneNumbers that takes in an array of phone number strings in (XXX) XXX-XXXX format and returns an array with the phone number strings in XXXXXXXXXX format.
+
+For example, (123) 456-7890 returns 1234567890
+------------------------------------------------------------------------------------------------ */
+
+const standardizePhoneNumbers = (arr) => {
+  let standardizedArr = [];
+  arr.forEach(num => {
+    standardizedArr.push(num.replace(/\(|\)|\s|-/g, '')
+    );
+  }
+  );
+  return standardizedArr;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 5
+
+Write a function named onlyOddChars that takes in a string and returns only the odd-index characters from that string.
+
+For example, 'abcdefg' returns 'bdf'
+------------------------------------------------------------------------------------------------ */
+
+const onlyOddChars = (str) => {
+  return str.replace(/(.[.]*)/g, '');
+};
```

Comment:
Hi there!
- For this challenge, we want to return the odd-index chars from the string.
- For this one, we try and use methods `includes()`, `substring()` or `charAt()` methods. Or...

- An approach might be to declare a new variable with a value of ' '. Then we can loop over the input string and check the index for each element. If that index is odd, we add it to the empty string variable. (consider using `%` operator)

- Another approach might be to declare a variable and set it to empty arr. We loop over string and add odd indexed chars this way. At the end, we turn the arr that holds the chars at odd indexes back to a string and return it.(Can use `%` operator) 👍